wgpu/src/lib.rs

```diff
@@ -3990,6 +4071,18 @@ impl Surface {
                     id,
                     data,
                     owned: false,
+
+                    // Todo: how do we specify these values correctly?
```

Comment:
We should store the surface configuration in the surface.

Reply:
Yes, I think so too, but unfortunately, the configure method (of the surface) has an immutable reference to self. We could use interior mutability but this feels like a hack. Do you have any advice on how to resolve this?

Reply:
The way we're doing it now is entirely a hack to begin with - we should be actually reaching into the underlying resources - but because we can't do that efficiently currently (see #2710) we're doing this at the surface level. I would just throw it in a mutex.

Reply:
Ok, this should work.